src/Sarif.UnitTests/SarifVersionOneToCurrentTests.cs

```diff
@@ -225,49 +235,56 @@ public void SarifTransformerTests_ToCurrent_OneRunWithFiles()
 
             SarifLog v2Log = TransformVersionOneToCurrent(v1LogText);
 
-            v2Log.Runs.Should().NotBeNull();
-            v2Log.Runs.Count.Should().Be(1);
-
-            IDictionary<string, FileData> files = v2Log.Runs[0].Files;
-
-            files.Should().NotBeNull();
-            files.Count.Should().Be(3);
-
-            FileData fileData = files["file:///home/list.txt"];
-
-            fileData.MimeType.Should().Be("text/plain");
-            fileData.Length.Should().Be(43);
-            fileData.FileLocation.Should().BeNull();
-            fileData.Contents.Should().NotBeNull();
-            fileData.Contents.Binary.Should().Be("VGhlIHF1aWNrIGJyb3duIGZveCBqdW1wcyBvdmVyIHRoZSBsYXp5IGRvZw==");
-            fileData.Hashes.Should().NotBeNull();
-            fileData.Hashes[0].Algorithm.Should().Be("sha-256");
-            fileData.Hashes[0].Value.Should().Be("d7a8fbb307d7809469ca9abcb0082e4f8d5651e46d3cdb762d02d0bf37c9e592");
-            fileData.Properties.Should().BeNull();
-
-            fileData = files["file:///home/buildAgent/bin/app.zip"];
-
-            fileData.MimeType.Should().Be("application/zip");
-            fileData.Length.Should().Be(0);
-            fileData.FileLocation.Should().BeNull();
-            fileData.Contents.Should().NotBeNull();
-            fileData.Contents.Binary.Should().BeNull();
-            fileData.Hashes.Should().BeNull();
-            fileData.Properties.Should().NotBeNull();
-            fileData.Properties["my_key"].SerializedValue.Should().Be("\"some value\"");
-
-            fileData = files["file:///home/buildAgent/bin/app.zip#/docs/intro.docx"];
-
-            fileData.MimeType.Should().Be("application/vnd.openxmlformats-officedocument.wordprocessingml.document");
-            fileData.ParentKey.Should().Be("file:///home/buildAgent/bin/app.zip");
-            fileData.Offset.Should().Be(17522);
-            fileData.Length.Should().Be(4050);
-            fileData.FileLocation.Should().NotBeNull();
-            fileData.FileLocation.Uri.Should().NotBeNull();
-            fileData.FileLocation.Uri.OriginalString.Should().Be("file:///docs/intro.docx");
-            fileData.Contents.Should().NotBeNull();
-            fileData.Contents.Binary.Should().BeNull();
-            fileData.Properties.Should().BeNull();
+            string v2LogText = JsonConvert.SerializeObject(v2Log, SarifTransformerUtilities.JsonSettingsV2);
+            string v2LogExpectedText =
+@"{
+  ""$schema"": ""http://json.schemastore.org/sarif-2.0.0"",
+  ""version"": ""2.0.0"",
+  ""runs"": [
+    {
+      ""tool"": {
+        ""name"": ""CodeScanner"",
+        ""semanticVersion"": ""2.1.0""
+      },
+      ""files"": {
+        ""file:///home/list.txt"": {
+          ""length"": 43,
+          ""mimeType"": ""text/plain"",
+          ""contents"": {
+            ""text"": ""VGhlIHF1aWNrIGJyb3duIGZveCBqdW1wcyBvdmVyIHRoZSBsYXp5IGRvZw=="",
```

Comment:
The MIME type tells us that this is a text file. In that case, the v1 spec tells us that v1 `file.contents` is a string containing a MIME-64 encoding of the UTF-8-transcoded contents of the file.

So we should populate v2 `file.contents.text` by decoding v1 `file.contents`. The result should be readable.

We don't know the original file's encoding, so we can't deduce the binary contents of the original file. So we should _not_ populate v2 `file.contents.binary`.

I could've sworn I did a review of that logic. Am I high?

Reply:
No you are probably right. I definitely made the change, let me go find it in my other branch.

Reply:
Fixed